Projects/UOContent/Items/TreasureChests/TreasureChestLevel3.cs

```diff
@@ -76,25 +76,35 @@ public TreasureChestLevel3() : base(0xE41)
         }
 
         // Equipment
-        for (var i = Utility.Random(3); i > 0; i--)
+        // Weapons
+        var weapon = Loot.RandomWeapon();
+        if (weapon != null)
+        {
+            weapon.DamageLevel = (WeaponDamageLevel)Utility.Random(3);
+            weapon.AccuracyLevel = (WeaponAccuracyLevel)Utility.Random(3);
+            weapon.DurabilityLevel = (WeaponDurabilityLevel)Utility.Random(3);
+            weapon.Quality = WeaponQuality.Regular;
+            DropItem(weapon);
+        }
+        
+        // Armor
+        var armor = Loot.RandomArmor();
+        if (armor != null)
+        {
+            armor.ProtectionLevel = (ArmorProtectionLevel)Utility.Random(3);
+            armor.Durability = (ArmorDurabilityLevel)Utility.Random(3);
+            armor.Quality = ArmorQuality.Regular;
+            DropItem(armor);
+        }
+        
+        // Shield
+        var shield = Loot.RandomShield();
+        if (shield != null)
         {
-            var item = Loot.RandomArmorOrShieldOrWeapon();
```

Comment:
This should never return a null item. Do we have a bug somewhere or is this caution?